src/modules/leaderboard.ts

```diff
@@ -31,6 +31,51 @@ async function updateLeaderboardRow(address: string, property: string, value: nu
   return true;
 }
 
+async function updateLeaderboardNfts() {
+  const name = `updateLeaderboardNfts`;
+  logger.info(name);
+
+  try {
+    const proposals = await db.merge_proposals.findAll({
+      include: [
+        {
+          association: "issue",
+          required: true,
+          where: {
+            state: "closed",
+          },
+        },
+      ],
+    }).then(v => v.filter(({ contractId, issue: { merged }}) => contractId?.toString() === merged))
+
+    if (proposals.length === 0) {
+      logger.warn(name, `no merged proposals found`);
+      return;
+    }
+
+    const nftAddress: {
+      [index:string]: number
+    } = {}
+
+    for(const proposal of proposals){
```

Comment:
Instead of iterate over proposals and querying the distributions, you can include the association `proposal_distributions` to the query, group by the recipient and then count the proposals